stackit/internal/services/argus/instance/resource.go

```diff
@@ -784,6 +1092,174 @@ func mapMetricsRetentionField(r *argus.GetMetricsStorageRetentionResponse, model
 	return nil
 }
 
+func mapUpdateAlertConfigField(ctx context.Context, resp *argus.UpdateAlertConfigsResponse, model *Model) error {
+	if resp == nil || resp.Data == nil || resp.Data.Receivers == nil {
+		model.AlertConfig = types.ObjectNull(alertConfigTypes)
+		return nil
+	}
+
+	if model == nil {
+		return fmt.Errorf("nil model")
+	}
+
+	// Receivers
+	respReceivers := *resp.Data.Receivers
+	receiversList := []attr.Value{}
+
+	for i := range respReceivers {
+		receiver := respReceivers[i]
+
+		// emailConfigs
+		emailConfigList := []attr.Value{}
+		if receiver.EmailConfigs != nil {
+			for _, emailConfig := range *receiver.EmailConfigs {
+				emailConfigList = append(emailConfigList, types.ObjectValueMust(emailConfigsTypes, map[string]attr.Value{
```

Comment:
We shouldn't use `...ValueMust` function in the provider code, as they panic in case of an error in the values instead of returning an error. The function description also says: "This creation function is only recommended to create List values which will not potentially affect practitioners, such as testing, or exhaustively tested provider logic."

Reply:
I think we only use them in tests so far

Reply:
That was an oversight on my side, you are 100% right. Fixed now